authorpages/authorpages.py

```diff
@@ -0,0 +1,39 @@
+import os
+import os.path
+
+from pelican import signals
+
+
+def yield_files(root):
+    root = os.path.realpath(os.path.abspath(root))
+    for dirpath, dirnames, filenames in os.walk(root):
+        for dirname in list(dirnames):
+            try:
+                if dirname.startswith("."):
+                    dirnames.remove(dirname)
```

Comment:
If you mutate dirnames (at least by removing elements, possibly also by rearranging, etc) then os.walk respects those changes as it continues to iterate the file system. Here in just trying to skip dot-dirs (if any) for efficiency.